wfl/fit/ace.py

```diff
@@ -0,0 +1,299 @@
+        verbose=True, remote_info=None, wait_for_results=True):
+    """Runs ace_fit on a a set of fitting configs
+
+    Parameters
+    ----------
+    fitting_configs: ConfigSet_in
+        set of configurations to fit
+    ACE_name: str
+        name of ACE potential, saved into files with yace and json suffixes
+    params: dict
+        dict of keys to turn into command line for ace_fit. each key will be prepended by
+        '-' (single characer) or '--' (longer name)
+    ref_property_prefix: str, default 'REF\_'
+        string prefix added to atoms.info/arrays keys (energy, forces, virial, stress)
+    skip_if_present: bool, default True
+        skip fitting if output is already present
+    run_dir: str, default '.'
+        directory to run in
+    formats: list(str), default ['json', 'yace']
+        list of filename suffix formats to write potentials in
+    ace_fit_exec: str, default "wfl/scripts/ace_fit.jl"
+        executable for ace_fit
+    dry_run: bool, default False
+        do a dry run, which returns the matrix size, rather than the potential name 
+    verbose: bool, default True
+        print verbose output
+    remote_info: dict or wfl.pipeline.utils.RemoteInfo, or '_IGNORE' or None
+        If present and not None and not '_IGNORE', RemoteInfo or dict with kwargs for RemoteInfo
+        constructor which triggers running job in separately queued job on remote machine.  If None,
+        will try to use env var WFL_ACE_FIT_REMOTEINFO used (see below). '_IGNORE' is for
+        internal use, to ensure that remotely running job does not itself attempt to spawn another
+        remotely running job.
+    wait_for_results: bool, default True
+        wait for results of remotely executed job, otherwise return after starting job
+
+    Returns
+    -------
+    ace_file_base: str
+        base of output files, with various suffixes (in formats) added, as well as other files 
+        generated by ace_fit.jl
+
+    Environment Variables
+    ---------------------
+    WFL_ACE_FIT_REMOTEINFO: JSON dict or name of file containing JSON with kwargs for RemoteInfo
+        contructor to be used to run fitting in separate queued job
+    ACE_FIT_JULIA_THREADS: used to set JULIA_NUM_THREADS for ace_fit.jl, which will use julia multithreading (LSQ assembly)
+    ACE_FIT_BLAS_THREADS: used by ace_fit.jl for number of threads to set for BLAS multithreading in ace_fit
+    """
+    if run_dir is None:
+        run_dir = '.'
+    run_dir = Path(run_dir)
+
+    assert isinstance(ref_property_prefix, str) and len(ref_property_prefix) > 0
+
+    remote_info = to_RemoteInfo(remote_info, 'WFL_ACE_FIT_REMOTEINFO')
+
+    ace_file_base = str(run_dir / ACE_name)
+
+    def _read_size():
+        with open(ace_file_base + '.size') as fin:
+            fields = fin.readline().strip().split()
+            return (int(fields[3]), int(fields[5]))
+
+    # figure out whether fit calculations are needed
+    need_to_fit = True
```

Comment:
I don't like having the `need_to_fit` flag, could it do without? 

For example

```
    # figure out whether fit calculations are needed
    if skip_if_present:
        # see if we can open and parse each output file
        try:
            _... not showing some code ..._ 
                        warnings.warn('Cannot parse ACE file with suffix '+suffix)
            if dry_run:     # previous "dry_run and not need_to_fit" would have only been True if an exception hasn't been raised, so can do here
                return size
            else:
                return ace_file_base  # "not need to fit" would have only been True if exception hasn't been raised, do here
        except:
                pass   # couldn't read file, need to fit.    

    if remote_info is not None and remote_info != '_IGNORE': 
        ...
```

Reply:
It was originally needed when the code also did the fitting/testing error calculation, and there were two flags, one for the actual fit, one for error calculation.  Let me think about whether it can be removed now.

Reply:
refactored to clean up in 16cc346